### DISCUSSION TITLE
How do I automate discussion for every page?

### DISCUSSION THREAD

**Discussion by 404Nikhil** [giscus/giscus#1430] [Q&A] [answered]
I have an application with dynamic routes like xyz/[id], and I want to integrate Giscus under each post. However, I currently have to manually create a discussion page for each blog post, which is not scalable.

**Comment by laymonage** [comment]
You don’t have to. New discussions will automatically be created when someone leaves the first comment.

**Reply by 404Nikhil** [reply]
i'm facing this error comments worked when i manually created the discussion 
![image](https://github.com/user-attachments/assets/b8538cdf-3119-4e28-b5dd-e6b7f876ffbd)

**Reply by laymonage** [reply]
You probably didn’t set up the repo ID and/or category ID correctly.

**Reply by 404Nikhil** [reply]
![image](https://github.com/user-attachments/assets/1bbd8d22-ae93-4c9e-a232-c6a83d9d9cfa)

hmm it works on https://anubhav-nine.vercel.app/blog/66712d935447b0bae4f5e3fd since i have manually created the discussion for this blog/[id] but for another blog page for example https://anubhav-nine.vercel.app/blog/6693ce4a0eefa12387868dc4 on adding comments it's showing these errors in console. Is there a proper documentation of integrating giscus in react i think i'm missing this addDiscussionComment promise

**Reply by laymonage** [reply]
It’s `repoId`, not `repo-id`. You don’t need `src`.

**Reply by 404Nikhil** [reply]
![image](https://github.com/user-attachments/assets/da582234-a39c-4afc-8371-4ddb82298e7c)
![image](https://github.com/user-attachments/assets/86a1d158-a494-454e-b4c8-5837d7ae47c6)

still same issue is persisting

**Reply by laymonage** [reply, marked as answer]
The category ID doesn't seem to match the "Announcements" category you specified. I tried running your repo through the https://giscus.app configuration and I get the value `DIC_kwDOMJ3j1s4CgbtK`.

**Reply by 404Nikhil** [reply]
thank you so much @laymonage
